tests/tests_release_notes_workflow/test_component_release_notes.py

```diff
@@ -0,0 +1,81 @@
+
+
+class TestComponentReleaseNotes(unittest.TestCase):
+
+    @patch('subprocess.check_call', return_value=0)
+    @patch('subprocess.check_output', return_value='c3408f34961eb7e8ebc0510306f62238a457dbbc'.encode())
+    @patch("subprocess.check_output", return_value='.release-notes-2.2.0.0.md'.encode())
+    def test_plugin_check(self, *mocks: Any) -> None:
+        component = InputComponentFromSource({"name": "common-utils", "repository": "https://github.com/opensearch-project/common-utils", "ref": "c3408f34961eb7e8ebc0510306f62238a457dbbc"})
+        self.repo = GitRepository(
+            url="https://github.com/opensearch-project/common-utils",
+            ref="c3408f34961eb7e8ebc0510306f62238a457dbbc",
+        )
+        self.assertEqual(self.repo.url, "https://github.com/opensearch-project/common-utils")
+        self.assertEqual(self.repo.ref, "c3408f34961eb7e8ebc0510306f62238a457dbbc")
+        self.assertEqual(self.repo.sha, "c3408f34961eb7e8ebc0510306f62238a457dbbc")
+        release_notes = ComponentsReleaseNotes.from_component(component, "2.2.0", self.repo.dir)
+        self.assertEqual(release_notes.filename, '.release-notes-2.2.0.0.md')
+
+    @patch('subprocess.check_call', return_value=0)
+    @patch('subprocess.check_output', return_value='a081f2f13e5df63dcc0897ded2154d7eab3f306a'.encode())
+    @patch("subprocess.check_output", return_value='.release-notes-2.2.0.0.md'.encode())
+    def test_core_check(self, *mocks: Any) -> None:
+        component = InputComponentFromSource({"name": "OpenSearch", "repository": "https://github.com/opensearch-project/OpenSearch", "ref": "a081f2f13e5df63dcc0897ded2154d7eab3f306a"})
+        self.repo = GitRepository(
+            url="https://github.com/opensearch-project/OpenSearch",
+            ref="a081f2f13e5df63dcc0897ded2154d7eab3f306a",
+        )
+        self.assertEqual(self.repo.url, "https://github.com/opensearch-project/OpenSearch")
+        self.assertEqual(self.repo.ref, "a081f2f13e5df63dcc0897ded2154d7eab3f306a")
+        self.assertEqual(self.repo.sha, "a081f2f13e5df63dcc0897ded2154d7eab3f306a")
+        release_notes = ComponentsReleaseNotes.from_component(component, "2.2.0", self.repo.dir)
+        self.assertEqual(release_notes.filename, '.release-notes-2.2.0.md')
+
+    @patch('subprocess.check_call', return_value=boolean)
+    @patch('subprocess.check_output', return_value='a081f2f13e5df63dcc0897ded2154d7eab3f306a'.encode())
+    @patch('release_notes_workflow.component_release_notes.ComponentReleaseNotesOpenSearchPlugin.path_exists')
+    @patch('release_notes_workflow.component_release_notes.ComponentReleaseNotesOpenSearchPlugin.path')
+    @patch('release_notes_workflow.component_release_notes.ComponentReleaseNotesOpenSearchPlugin.exists')
+    @patch('release_notes_workflow.component_release_notes.ComponentReleaseNotesOpenSearchPlugin.check')
+    def test_ComponentReleaseNotesOpenSearchPlugin(self, *mocks: Any) -> None:
```

Comment:
Move to `TestComponentReleaseNotesOpenSearchPlugin` and call it `test_exists`, separate tests for `test_check` and `test_path_exists`.

Reply:
Modified the tests.